docs/developer-docs/setup/pocket-ic.md

```diff
@@ -0,0 +1,278 @@
+# Testing canisters with PocketIC
+
+## Overview
+
+PocketIC is a lightweight, deterministic testing platform for programmatic testing for canisters written in Rust and Python. It seamlessly integrates with existing testing infrastructure, such as `cargo test`, and runs as a standalone binary that doesn't require additional containers or virtual machines. Currently, PocketIC is supported on MacOS and Linux host systems. 
+
+While there are other options for testing canisters, they may come with pitfalls such as:
+
+- Installing and testing canisters on the mainnet provides the 'real' workflow experience, but will cost developers real cycles.
+
+- Testing using the local replica provided by `dfx` allows for testing on a single IC node, but there is no cross-net or multi-subnet functionality. Testing with the local replica is not deterministic, and it is rather heavyweight.
```

Comment:
I would say additionally from my experience writing automated tests with DFX that it is complex, _slow_, requires quite a bit of boilerplate code to get it working, you have little control over the testing environment and it can interfere with your local testing environment if you don't automate it correctly.